LEAF_Request_Portal/templates/reports/LEAF_inbox_combined.tpl

```diff
@@ -1,337 +1,609 @@
+            if (Object.keys(res).length == batchSize) {
+                return loadInboxData(site, offset);
+            }
+        });
+    }
+
+    function buildWorkflowCategoryCache(site) {
+        return $.ajax({
             type: 'GET',
-            url: siteURL,
-            success: function (res) {
-                dataInboxes[site] = res;
-                sitesLoaded.push(site);
-            },
-            error: function (err) {
-                alert('Error: ' + err.statusText);
-            },
-            cache: false,
-            timeout: 5000
+            url: site.url + 'api/workflow/categories?x-filterData=categoryID',
+            success: function(res) {
+                res.forEach(w => {
+                    dataWorkflowCategories[w.categoryID] = 1;
+                });
+            }
         });
     }
 
-    function loadWorkflow(recordID, dependencyID, prefixID, rootURL) {
+    function loadWorkflow(recordID, prefixID, rootURL) {
         dialog_message.setTitle('Apply Action to #' + recordID);
         currRecordID = recordID;
         dialog_message.setContent('<div id="workflowcontent"></div><div id="currItem"></div>');
         workflow = new LeafWorkflow('workflowcontent', '<!--{$CSRFToken}-->');
         workflow.setRootURL(rootURL);
-        workflow.setActionSuccessCallback(function () {
+        workflow.setActionSuccessCallback(function() {
             dialog_message.hide();
             $('#' + prefixID + 'tbody_tr' + recordID).fadeOut(1500);
         });
         workflow.getWorkflow(recordID);
         dialog_message.show();
     }
 
-    // Polyfill for IE
-    if (typeof Object.assign != 'function') {
-        // Must be writable: true, enumerable: false, configurable: true
-        Object.defineProperty(Object, "assign", {
-            value: function assign(target, varArgs) { // .length of function is 2
-                'use strict';
-                if (target == null) { // TypeError if undefined or null
-                    throw new TypeError('Cannot convert undefined or null to object');
-                }
-                let to = Object(target);
-                for (let index = 1; index < arguments.length; index++) {
-                    let nextSource = arguments[index];
-                    if (nextSource != null) { // Skip over if undefined or null
-                        for (let nextKey in nextSource) {
-                            // Avoid bugs when hasOwnProperty is shadowed
-                            if (Object.prototype.hasOwnProperty.call(nextSource, nextKey)) {
-                                to[nextKey] = nextSource[nextKey];
-                            }
-                        }
-                    }
+    const getMapSites = new Promise((resolve, reject) => {
+        $.ajax({
+            type: 'GET',
+            url: './api/site/settings/sitemap_json',
+            success: function(res) {
+                if (res === 'Admin access required') {
```

Comment:
This comparison worries me.  Not only does it have capitols, but spaces as well.

Reply:
In the php file, if the user isn't in the admin group, it returns this specific message. I can alter it to return an int instead if that would be better.